### PR TITLE
Code blocking

### LINKED ISSUE #51: Incorrect path in PEP to the downloaded files
After downloading processed data with metadata, path to each file is incorrect. Project uploaded in peppy and pepr is have incorrect paths to the files. 
Initial yaml file has to be changed.

### PR DESCRIPTION
**Pull request by nsheff**
This PR:
- Fixes #51

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -2,21 +2,28 @@
 
 [![PEP compatible](http://pepkit.github.io/img/PEP-compatible-green.svg)](http://pepkit.github.io)
 
-`geofetch` is a command-line tool that downloads and organizes data and metadata from GEO and SRA. When given one or more GEO/SRA accessions, `geofetch` will:
+`geofetch` is a command-line tool that downloads and organizes data and metadata from GEO and SRA. When given one or more GEO/SRA accessions, `geofetch` is:
```
Comment (nsheff) [inline docs/README.md]: incorrect

File: docs/changelog.md [hunk 2]
```diff
@@ -1,5 +1,16 @@
 # Changelog
 
+## [1.0.0] -- 2022-02-25
```
Comment (nsheff) [inline docs/changelog.md:3]: We should use a consistent versioning. please use `0.8.0`.

File: docs/usage.md [hunk 3]
```diff
@@ -1,54 +1,69 @@
...
-                        setting instead treats each run as a separate sample
+                        Split SRR runs into individual samples. By default, SRX experiments with multiple SRR
+                        Runs will have a single entry in the annotation table, with each run as a separate row
+                        in the subannotation table. This setting instead treats each run as a separate sample
   --config-template CONFIG_TEMPLATE
                         Project config yaml file template.
+  -k SKIP, --skip SKIP  Skip some accessions. [Default: no skip].
   -p, --processed       Download processed data [Default: download raw data].
+  --stored-in {all,samples,series}
+                        Optional: {processed} Specify GEO location where processed data is stored and has to
```
Comment (nsheff) [inline docs/usage.md:45]: I do not understand what is meant by `{processed}` in these help docs.

Comment (nsheff) [reply docs/usage.md:45]: `stored-in` needs to be renamed. Does it only apply when `--processed` is set?

File: docs_jupyter/row-data-downloading.ipynb [hunk 4]
```diff
@@ -4,9 +4,9 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# geofetch tutorial\n",
+    "# geofetch tutorial for raw (.sra) files\n",
     "\n",
-    "The [GSE67303 data set](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE67303) has about 250 mb of data across 4 samples, so it's a quick download for a test case. Let's take a quick peek at the geofetch version:"
+    "The [GSE150868 data set](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE150868) has about 250 mb of data across 4 samples, so it's a quick download for a test case. Let's take a quick peek at the geofetch version:"
```
Comment (nsheff) [inline docs_jupyter/row-data-downloading.ipynb:9]: This is wrong. Why did you change the example GSE? The one you replaced it with has 35 samples, but you say it has 4 samples. It also has ~ 50 Gb of data, not 250 mb of data, so it's a bad example. Why did you switch this?

File: docs_jupyter/row-data-downloading.ipynb [hunk 5]
```diff
@@ -18,7 +18,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "geofetch 0.7.0\n"
+      "geofetch 1.0.0\n"
```
Comment (nsheff) [inline docs_jupyter/row-data-downloading.ipynb:21]: Use 0.8.0.

File: geofetch/utils.py [hunk 6]
```diff
@@ -4,6 +4,7 @@
 import logging
 import os
 import subprocess
+import re
 
 
 __author__ = ["Vince Reuter", "Nathan Sheffield"]
```
Comment (nsheff) [inline geofetch/utils.py:10]: add yourself to author list?

File: mkdocs.yml [hunk 7]
```diff
@@ -8,10 +8,11 @@ nav:
   - Getting started:
     - Introduction: README.md
     - Install and configure: install.md
-    - Tutorial: tutorial.md
     - SRA convert: sra_convert.md
-  - How-to Guides:
+  - How-to Guides Tutorials:
```
Comment (nsheff) [inline mkdocs.yml:12]: this is not the right way to say this.

File: requirements/requirements-all.txt [hunk 8]
```diff
@@ -3,3 +3,6 @@ colorama>=0.3.9
 peppy>=0.19.1
 logmuse>=0.0.2
 ubiquerg>=0.4.4
+logmuse>=0.2.7
+ubiquerg>=0.6.0
+pandas>=1.4.0
```
Comment (nsheff) [inline requirements/requirements-all.txt:8]: will pandas be required here?

if you removed pandas don't forget to remove from dependencies

File: geofetch/geofetch.py [hunk 9]
```diff
@@ -24,502 +23,1081 @@
...
-            default=False,
-            action="store_true",
-            help="Download processed data [Default: download raw data].")
+        # if user specified a pipeline interface path, add it into the project config
+        if self.args.pipeline_interfaces:
+            self.file_pipeline_interfaces = self.args.pipeline_interfaces
+        else:
+            self.file_pipeline_interfaces = "null"
+
+    def run_geofetch(self):
```
Comment (nsheff) [inline geofetch/geofetch.py:125]: maybe just `fetch` ?

File: geofetch/geofetch.py [hunk 10]
```diff
@@ -24,502 +23,1081 @@
...
+SAMPLE_SUPP_METADATA_FILE = "_annotation_sample_processed.csv"
+EXP_SUPP_METADATA_FILE = "_annotation_series_processed.csv"
+
 # How many times should we retry failing prefetch call?
 NUM_RETRIES = 3
 
 
-def _parse_cmdl(cmdl):
-    parser = argparse.ArgumentParser(description="Automatic GEO and SRA data downloader")
+class Geofetch:
```
Comment (nsheff) [inline geofetch/geofetch.py:61]: Maybe `Geofetcher` ?

File: geofetch/geofetch.py [hunk 11]
```diff
@@ -24,502 +23,1081 @@
...
+
+        acc_GSE_list = parse_accessions(self.args.input, self.metadata_expanded, self.args.just_metadata)
+
+        # Loop through each accession.
+        # This will process that accession, produce metadata and download files for
+        # the GSM #s included in the list for each GSE#.
+        # acc_GSE = "GSE61150" # example
+
+        # This loop populates a list of metadata.
+        metadata_dict = OrderedDict()
```
Comment (nsheff) [inline geofetch/geofetch.py:136]: After python 3.6, all dicts are ordered... so if we drop support for python <3.6, these can just become built-in dicts and we can clean this up.

File: geofetch/geofetch.py (hunk 9 above)
Comment (nsheff) [reply geofetch/geofetch.py:125]: or `fetch_all` ?

**Review by nsheff — COMMENTED** [review]
Can you also make sure to run `black` on this?

File: docs/usage.md (hunk 3 above)
Comment (Khoroshevskyi) [reply docs/usage.md:45]: I have added "{processed} " to description, it means that this argument will have any effect only when processed data or metadata has been collecting.
Do you have any suggestions for new name for stored-in argument?

File: docs_jupyter/row-data-downloading.ipynb (hunk 4 above)
Comment (Khoroshevskyi) [reply docs_jupyter/row-data-downloading.ipynb:9]: I will find better series and change it.

File: geofetch/geofetch.py (hunk 11 above)
Comment (Khoroshevskyi) [reply geofetch/geofetch.py:136]: Do you suggest to change all OrderedDict to just dict? OrderedDict was used in previous versions and I didn't change it.

File: requirements/requirements-all.txt (hunk 8 above)
Comment (Khoroshevskyi) [reply requirements/requirements-all.txt:8]: Geofetch is not using pandas, I will remove it.

File: docs_jupyter/row-data-downloading.ipynb (hunk 4 above)
Comment (nsheff) [reply docs_jupyter/row-data-downloading.ipynb:9]: what was wrong with the original example?

Comment (Khoroshevskyi) [reply docs_jupyter/row-data-downloading.ipynb:9]: I wanted to find some series that has raw data, series and samples processed data. But I something went wrong. I think for raw data we can use previous example, and for processed data create new file.

Comment (nsheff) [reply docs_jupyter/row-data-downloading.ipynb:9]: ok.

File: docs/usage.md (hunk 3 above)
Comment (nsheff) [reply docs/usage.md:45]: You have to actually say what you mean, the user isn't going to be able to understand that. I think the name could be `processed-data-source` or just `data-source`

Comment (nsheff) [reply docs/usage.md:45]: The message can read something like:

> Specifies the source of data on the GEO record to retrieve processed data, which may be attached to the collective series entity, or to individual samples. Ignored unless 'processed' flag is set.

File: geofetch/geofetch.py (hunk 11 above)
Comment (nsheff) [reply geofetch/geofetch.py]: It looks like you made this switch, but there's still a lot of references to OrderedDict in the docstrings -- can you correct those?

Comment (nsheff) [reply geofetch/geofetch.py]: Also, how did you specify the minimum python version?

Comment (Khoroshevskyi) [reply geofetch/geofetch.py]: > Also, how did you specify the minimum python version?

I specified python 3.8 because I was using it while making changes to geofetch. Shoud I specify the minimum version 3.10?

**Comment by nsheff** [comment]
> #53 Fixed

great, thanks!

File: geofetch/geofetch.py (hunk 11 above)
Comment (Khoroshevskyi) [reply geofetch/geofetch.py]: > It looks like you made this switch, but there's still a lot of references to OrderedDict in the docstrings -- can you correct those?

Corrected !

**Comment by nsheff** [comment]
> #53 Fixed

I don't think it works like this. I think it has to be "fixes #X", not "#X fixed". See https://github.blog/2013-01-22-closing-issues-via-commit-messages/

File: geofetch/geofetch.py [hunk 12]
```diff
@@ -591,7 +607,7 @@ def expand_metadata_list(self, metadata_list, dict_key):
                             metadata_list[n_elem].update(sample_char)
                         else:
                             just_string = True
-                            this_string += elem
+                            this_string += ", " + elem
```
Comment (nsheff) [inline geofetch/geofetch.py:610]: use `join` instead of `+` for this kind of thing. See: https://www.programiz.com/python-programming/methods/string/join

Comment (Khoroshevskyi) [reply geofetch/geofetch.py]: fixed

**Comment by nsheff** [comment]
this is great, nice work @Khoroshevskyi !
